### PR TITLE
Inject CSRF token into template to make it work with CSRF_COOKIE_HTTPONLY

### PR DESCRIPTION
**Pull request by Duologic**
Discussion in issue: https://github.com/YPlan/nexus/issues/90

### CODE REVIEW

File: nexus/media/js/nexus.js [hunk 1]
```diff
@@ -34,8 +19,8 @@ jQuery.ajaxSetup({
         }
 
         if (!safeMethod(settings.type) && sameOrigin(settings.url)) {
-            var cookieName = $('#nexus-constants').data('csrfCookieName');
-            xhr.setRequestHeader("X-CSRFToken", getCookie(cookieName));
+            var csrf_token = $('#nexus-constants').data('csrfToken');
```
Comment (graingert) [inline nexus/media/js/nexus.js]: csrfToken

**Review by graingert — CHANGES_REQUESTED** [review]

File: nexus/media/js/nexus.js (hunk 1 above)
Comment (graingert) [inline nexus/media/js/nexus.js]: or CSRFToken

File: nexus/templates/nexus/base.html [hunk 2]
```diff
@@ -19,7 +19,7 @@
         <script src="{% nexus_media_prefix %}/nexus/js/lib/facebox/facebox.js"></script>
         <script src="{% nexus_media_prefix %}/nexus/js/nexus.js"
                 id="nexus-constants"
-                data-csrf-cookie-name="{% nexus_csrf_cookie_name %}"
+                data-csrf-token="{{ csrf_token }}"
```
Comment (graingert) [inline nexus/templates/nexus/base.html:22]: tbh, I'd probably go for data-nexus-csrf-token to match data-nexus-media-prefix, but up to @adamchainz

**Review by graingert — CHANGES_REQUESTED** [review]

File: nexus/templates/nexus/base.html (hunk 2 above)
Comment (Duologic) [reply nexus/templates/nexus/base.html:22]: Sounds sane, I'll keep it open for now.

**Comment by graingert** [comment]
@Duologic also update the HISTORY.rst

**Comment by graingert** [comment]
Commit message: SSL -> TLS, although it's redundant here HttpOnly is what matters.
Add a `Fixes: #90` somewhere.

**Comment by Duologic** [comment]
I'll squash the changes into one commit

**Comment by Duologic** [comment]
Thanks for the review!

File: nexus/templates/nexus/base.html (hunk 2 above)
Comment (adamchainz) [reply nexus/templates/nexus/base.html:22]: no need for 'nexus' in name imo, the nexus in nexus-media-prefix was pre-existing for the `nexus_media_prefix` tag

File: nexus/templatetags/nexus_helpers.py [hunk 3]
```diff
@@ -21,11 +20,6 @@ def nexus_version():
     return nexus.__version__
 
 
-@register.simple_tag
-def nexus_csrf_cookie_name():
```
Comment (adamchainz) [inline nexus/templatetags/nexus_helpers.py:25]: I'd document the removal of this in HISTORY.rst just in case someone has used it in their own templates

**Review by adamchainz — CHANGES_REQUESTED** [review]

File: nexus/templatetags/nexus_helpers.py (hunk 3 above)
Comment (Duologic) [reply nexus/templatetags/nexus_helpers.py:25]: Done.

**Comment by Duologic** [comment]
Should I change the version in `nexus/__init__.py` or does that happen in the merge?

**Comment by adamchainz** [comment]
That happens in release commits only

**Comment by Duologic** [comment]
Alright, I think I got everything.

How do we go from here? Do you guys have release cycle of some kind?

**Comment by adamchainz** [comment]
I'll merge and release tonight

**Comment by Duologic** [comment]
Long days you have there... :P

**Comment by adamchainz** [comment]
Yes I'm on Venus

**Review by adamchainz — APPROVED** [review]

**Comment by adamchainz** [comment]
Merged in 73375ec0ede7315049c16cdce6bfdd9ed32cd19b, thanks!

**Comment by adamchainz** [comment]
V1.5.0 on PyPI now: https://pypi.python.org/pypi/nexus-yplan

**Comment by Duologic** [comment]
Closing the feedback loop here, this is now in production for us and working terrific. Thanks for maintaining the nexus and gargoyle packages!

**Pull request closed without merging**